draco/analysis/beam.py

```diff
@@ -69,48 +73,50 @@ def process(self, data, beam):
 
         freq = beam.freq[fstart:fstop]
 
-        # Make sure the beam is in celestial coordinates
-        if beam.coords != "celestial":
-            raise RuntimeError(
-                "Beam must be converted to celestial coordinates prior to generating "
-                "a HybridVisStream."
-            )
+        # Extract the coordinates
+        el_data = data.index_map["el"]
 
-        # Check that el matches
-        dec = beam.theta
+        if beam.coords == "celestial":
+            dec = np.radians(beam.theta)
+            el_beam = np.sin(dec - self.latitude)
 
-        el_beam = np.sin(np.radians(dec - self.telescope.latitude))
-        el_data = data.index_map["el"]
+            ha = beam.phi
+            ra = (ha + 360.0) % 360.0
+            nra = int(round(360.0 / np.abs(ha[1] - ha[0])))
+            delta_ra = 360.0 / nra
```

Comment:
```suggestion
            ra_beam = (ha + 360.0) % 360.0
            nra = int(round(360.0 / np.abs(ha[1] - ha[0])))
            delta_ra = 360.0 / nra
            ra = np.arange(nra) * delta_ra
```

With this change (and propagating `ra_beam` in the next few lines) the code ran to completion and I confirmed that the result is the same as with the previous version by comparing the ringmap and weights for about 20 unflagged randomly selected frequencies in the lower band.